index.bs

```diff
@@ -569,13 +582,21 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered sets=].
 
-1. For each |authenticator| currently available on this platform, if
-    <code>|options|.{{MakeCredentialOptions/attachment}}</code> is [=present|not present=] or its value matches
-    |authenticator|'s [=attachment modality=], [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|.
-
-1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is
+1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is 
     "{{NotFoundError}}", and terminate this algorithm.
 
+1. If |options|.{{MakeCredentialOptions/authenticatorSelection}} is [=present|present=], iterate through 
+        |currentlyAvailableAuthenticators| and doing the following for each [=authenticator=]:
+    1. If |attachment| is [=present|not present=] or its value matches |authenticator|'s attachment modality, continue
+        to the next step. Otherwise, [=iteration/continue=] to the next [=authenticator=].
+    1. If |requireResidentKey| is set to |true| and the |authenticator| is capable of storing 
+        [=Client-Side-Resident Credential Private Key=], continue to the next step. Otherwise, return a {{DOMException}} 
+        whose name is "{{QuotaExceededError}}" and [=iteration/continue=] to the next [=authenticator=]. 
```

Comment:
I was thinking we can send one DOMException but not breaking the algorithm. In another word, this will let developer know one authenticator on the platform doesn't have the storage. If there's another authenticator that does have the storage, then the authentication will still work. Is that not the right web behavior? I can also remove the QuotaExceededError.